helix-term/src/ui/picker.rs

```diff
@@ -349,7 +349,7 @@ impl<T> Picker<T> {
     pub fn score(&mut self) {
         let now = Instant::now();
 
-        let pattern = self.prompt.line();
+        let pattern = &self.prompt.line().trim_start().to_string();
```

Comment:
The `to_string()` allocation is only necessary for line 422, which does another `clone`. I would remove here the `&...to_string()` and put in 422 `self.previous_pattern = pattern.to_string();`